test/test_helper.rb

```diff
@@ -10,54 +10,73 @@
 
 # Docs for our Webview lib: https://github.com/Maaarcocr/webview_ruby
 
-SCARPE_EXE = File.expand_path("../exe/scarpe", __dir__)
+def with_tempfile(prefix, contents)
+  t = Tempfile.new(prefix)
+  t.write(contents)
+  t.flush # Make sure the contents are written out
+
+  yield(t.path)
+ensure
+  t.close
+  t.unlink
+end
 
-TEST_OPTS = [:timeout, :allow_fail, :debug]
-def test_scarpe_app(body_code, opts = {})
+SCARPE_EXE = File.expand_path("../exe/scarpe", __dir__)
+TEST_OPTS = [:timeout, :allow_fail, :debug, :exit_immediately]
+def test_scarpe_app(body_code, test_code: "", **opts)
   bad_opts = opts.keys - TEST_OPTS
   raise "Bad options passed to test_scarpe_app: #{bad_opts.inspect}!" unless bad_opts.empty?
 
-  out = Tempfile.new("scarpe_test_results.json")
-  out_path = File.expand_path out.path
-  Tempfile.open("scarpe_test") do |f|
-    do_debug = opts[:debug] ? true : false
-    die_after = opts[:timeout] ? opts[:timeout].to_f : 1.0
-    f.write(
-      "Scarpe.app(test_assertions: true, debug:#{do_debug.inspect}, die_after: #{die_after}, " \
-        "result_filename: #{out_path.inspect}) do\n",
-    )
-    f.write(body_code)
-    f.write("\nend\n")
-    f.flush # Make sure the code is written out
-
-    script_location = File.expand_path(f.path)
-    system("ruby #{SCARPE_EXE} --dev #{script_location}")
-    f.unlink
-  end
+  do_debug = opts[:debug] ? true : false
+  die_after = opts[:timeout] ? opts[:timeout].to_f : 1.0
+  scarpe_app_code = <<~SCARPE_APP_CODE
+    Scarpe.app do
+      #{body_code}
+    end
+  SCARPE_APP_CODE
 
-  # If failure is okay, don't check for status or assertions
-  return if opts[:allow_fail]
+  result_path = nil
```

Comment:
It seems like this doesn't get called again in this scope, can it be removed?

Reply:
Ah, good catch! Yes.